server/src/model/user.py

```diff
@@ -49,24 +50,26 @@ def add_google_user(token: str,) -> Tuple[Optional[AppError], Optional[User]]:
 
         existing_user = User.lookup_google_user(user_info["id"])
         if existing_user:
-            # remove after unassigned_family migration
-            if "unassigned_family_id" not in existing_user:
-                family_error, unassigned_family = Family.add_new_family(
-                    "Unassigned", UNASSIGNED_FAMILY_INTERVAL
-                )
-                if family_error or not unassigned_family:
-                    return (family_error, None)
-                existing_user.unassigned_family_id = unassigned_family.id
-
             existing_user.google_token = token
             existing_user.save()
             return (None, existing_user)
 
-        family_error, unassigned_family = Family.add_new_family(
-            "Unassigned", UNASSIGNED_FAMILY_INTERVAL
+        # Create Unassigned family
+        unassigned_family_error, unassigned_family = Family.add_new_family(
+            UNASSIGNED_FAMILY.name, UNASSIGNED_FAMILY.sync_interval_days
         )
-        if family_error or not unassigned_family:
-            return (family_error, None)
+        if unassigned_family_error or not unassigned_family:
+            return (unassigned_family_error, None)
+
+        # Create starter families
+        family_ids: List[ObjectId] = [unassigned_family.id]
+        for starter_family in STARTER_FAMILIES:
+            family_error, family = Family.add_new_family(
```

Comment:
lets change the method signature of Family.add_new_family( to accept the NewFamily dataclass.

Reply:
Going that route would add a layer of complexion that I think is unnecessary for now. For example, when we start interfacing with client on family manipulation, we would have to wrap the payload to the dataclass. Let’s revisit that later.

Reply:
would typed dict be better then?
So as to follow what we do for NewMatesInput

Reply:
`NewMatesInput` is a bit of a different case. Personally, I don't think that was ideal, but that was needed due to interfacing with GraphQL (need to define a new GQL type when working with non-native primitives). I think it would be good to leave the creation function args as primitives when possible to not couple the API and data layer too much? It would be nice to overload the fn with these different use cases, but that's not possible with python.